eureka-client/src/main/java/com/netflix/discovery/endpoint/DnsResolver.java

```diff
@@ -33,15 +33,10 @@
     private static final String CNAME_RECORD_TYPE = "CNAME";
     private static final String TXT_RECORD_TYPE = "TXT";
 
-    static final DirContext dirContext = getDirContext();
-
-    private DnsResolver() {
-    }
-
     /**
      * Load up the DNS JNDI context provider.
      */
-    public static DirContext getDirContext() {
+    private DirContext getDirContext() {
```

Comment:
Also this is now called N+1 times, where N is number of zones. Are we okay with this? Isn't it time-consuming? (IMHO even if it is, it's not critical, but maybe somebody has another opinion).

Reply:
it's not an expensive request, we don't make a network call, just initialize context where it re-check dns server on the local machine.